redis-util/src/main/java/org/sunbird/redis/RedisConnectionManager.java

```diff
@@ -0,0 +1,143 @@
+        initialisingClusterServer(host, port);
+        return true;
+      }
+      ProjectLogger.log("RedisConnectionManager:initialiseConnection initalising Single Server.",
+          LoggerEnum.INFO.name());
+      initialiseSingleServer(host, port);
+      return true;
+    } catch (Exception e) {
+      ProjectLogger.log("Error while initialising elastic search connection", e);
+      return false;
+    }
+  }
+
+  private static void initialiseSingleServer(String host, String port) {
+    Config config = new Config();
+    config.useSingleServer().setAddress(host + ":" + port);
+    config.setCodec(new StringCodec());
+    client = Redisson.create(config);
+
+  }
+
+  // cluster state scan interval in milliseconds
+  // use "rediss://" for SSL connection
+  private static void initialisingClusterServer(String host, String port) {
+    String[] hosts = host.split(",");
+    String[] ports = port.split(",");
+    Config config = new Config();
+    try {
+      config.setCodec(new StringCodec());
+      ClusterServersConfig clusterConfig = config.useClusterServers();
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer with ip :" + hosts.toString(),
+          LoggerEnum.INFO.name());
+
+      clusterConfig.setScanInterval(2000);
+      for (int i = 0; i < hosts.length && i < ports.length; i++) {
+        clusterConfig.addNodeAddress("redis://" + hosts[i] + ":" + ports[i]);
+      }
+
+      client = Redisson.create(config);
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer client created ", LoggerEnum.INFO.name());
+    } catch (Exception e) {
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer Exception " + e, LoggerEnum.INFO.name());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public static void initializeSectionCache() {
+    try {
+      RedisCacheUtil.flushCache(ActorOperations.GET_SECTION.getValue());
+      Response response = cassandraOperation.getAllRecords(KEY_SPACE_NAME, PAGE_SECTION_TABLE_NAME);
+      List<Map<String, Object>> responseList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+      if (null != responseList && !responseList.isEmpty()) {
+        for (Map<String, Object> resultMap : responseList) {
+
+          RedisCacheUtil.addDataToRedisMap(ActorOperations.GET_SECTION.getValue(), (String) resultMap.get(JsonKey.ID),
+              redisUtil.getJsonString(resultMap));
+        }
+      }
+      ProjectLogger.log("RedisCacheInitialiser:initializeSectionCache " + PAGE_SECTION_TABLE_NAME + " cache size: "
+          + responseList.size(), LoggerEnum.INFO.name());
+
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "RedisCacheInitialiser:initializeSectionCache: Exception in retrieving page section " + e.getMessage(), e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void initializePageCache() {
+    try {
+      RedisCacheUtil.flushCache(ActorOperations.GET_PAGE_DATA.getValue());
```

Comment:
RedisCache implementation should be agnostic of the usage. PageCache specific logic should be part of page management service / actor.